---
layout: rule
title: K1001
description: Message properties must be read-only.
permalink: /rules/1001
---

All messages must be immutable.  See [Message](/reference/message) for more information.  Remove the setter 
from all properties or set the setter to private.


### Noncompliant Code Example

The following example has a getter and setter.

{% highlight csharp %}
public class HelloWorldRequest
{
    // This property has beth get and set defined.
    public string Name { get; set;  }
}
{% endhighlight %}

### Compliant Examples

The following example uses a get only property.

{% highlight csharp %}
public class SomeRequest
{
    // Initialization moved to constructor which is required for serialization.
    public SomeRequest(string name)
    {
        this.Name = name;
    }

    // Property has getter only.
    public string Name { get; }
}
{% endhighlight %}

The following example has a private set.

{% highlight csharp %}
public class SomeRequest
{
    // Initialization moved to constructor which is required for serialization.
    public SomeRequest(string name)
    {
        this.Name = name;
    }

    // Property has a private set.  Some frameworks require there to be a setter.
    public string Name { get; private set; }
}
{% endhighlight %}

### Exceptions

None
