---
layout: default
title: K1000
description: Messages should not contain fields.
permalink: /rules/1000
---

Fields in messages indicate that there might be some logic when there shouldn't be.  Fields also are not serialized by default.  Remove any fields from message
classes or replace them with properties.


### Noncompliant Code Example

{% highlight csharp %}
public class SomeRequest
{
    private string _name;
}
{% endhighlight %}

### Compliant Solution
{% highlight csharp %}
public class SomeRequest
{
    public SomeRequest(string name)
    {
        this.Name = name;
    }

    public string Name { get; }
}
{% endhighlight %}

### Exceptions

None
